.github/workflows/cicd-pipeline.yml

```
@@ -316,23 +311,59 @@ jobs:
 
       - name: Delete Merged Release or Hotfix Branch
         if: steps.create_tag.outputs.should_run == 'true'
-        run: |
-          echo "Deleting merged release/hotfix branch: $SOURCE_BRANCH"
-          git push origin --delete $SOURCE_BRANCH
-        env:
-          SOURCE_BRANCH: ${{ github.event.pull_request.head.ref }}
+        uses: actions/checkout@v5
+        with:
+          ssh-key: ${{ secrets.DEPLOY_KEY_PRIVATE }}
+      - run: |
```

actionlint (1.7.8)

"github.event.pull_request.head.ref" is potentially untrusted. avoid using it directly in inline scripts. instead, pass it through an environment variable. see https://docs.github.com/en/actions/reference/security/secure-use#good-practices-for-mitigating-script-injection-attacks for more details

(expression)